### ISSUE TITLE
Meson setup warning in subprojects when setting language standard using 'default_options'

### ISSUE THREAD

**Issue by killekulla** [mesonbuild/meson#7810]
When configuring a project containing subprojects, and the top level meson.build is defining '`default_options`' with some language standard, the `meson setup` command issues a warning:

**|WARNING: In subproject sub: Unknown options: "sub:cpp_std"**

The warning message exists since #6597, bisected to 591e6e94b9fccfc49ee7093cb21735a27fd64005 (@dcbaker)

**To Reproduce**
_Project structure:_
```
. 
├── meson.build
└── subprojects
    └── sub
        └── meson.build
```
_./meson.build:_
```
project('test',	'cpp', default_options : ['cpp_std=c++17'])
subproject('sub')
```
_./subprojects/sub/meson.build:_
```
project('sub')
```

__Command to reproduce the warning:__
```
meson setup build
```

__Complete output:__
```
The Meson build system
Version: 0.55.999
Source dir: /home/rz/dev/temporary_ecs_group/test-meson
Build dir: /home/rz/dev/temporary_ecs_group/test-meson/build
Build type: native build
Project name: test
Project version: undefined
C++ compiler for the host machine: ccache c++ (gcc 6.3.0 "c++ (Debian 6.3.0-18+deb9u1) 6.3.0 20170516")
C++ linker for the host machine: c++ ld.bfd 2.35
Host machine cpu family: x86
Host machine cpu: i686

|Executing subproject sub method meson 
|
|WARNING: In subproject sub: Unknown options: "sub:cpp_std"
|The value of new options can be set with:
|meson setup <builddir> --reconfigure -Dnew_option=new_value ...
|Project name: sub
|Project version: undefined
|Build targets in project: 0
|Subproject sub finished.

Build targets in project: 0

test undefined

  Subprojects
    sub: YES

Found ninja-1.9.0 at /usr/bin/ninja
```

**Expected behavior**
Meson shall issues no warning message

**system parameters**
* Python 3.5.3
* meson version 0.55.999 (current master branch)

**Comment by xclaesse** [comment]
Right, I signaled that regression 2 months ago: https://github.com/mesonbuild/meson/pull/6597#issuecomment-671043489. It is fixed by my PR https://github.com/mesonbuild/meson/pull/7574, but @dcbaker refuse to agree on my solution.

**Comment by killekulla** [comment]
As of version 0.56.0 this warning is fixed by commit 7902d20. Thanks @xclaesse
